flow/CompressionUtils.cpp

```diff
@@ -110,27 +85,17 @@ StringRef CompressionUtils::decompress(const CompressionFilter filter, const Str
 	if (filter == CompressionFilter::NONE) {
 		return StringRef(arena, data);
 	}
-
-	namespace bio = boost::iostreams;
-	std::stringstream compStream(data.toString());
-	std::stringstream decompStream;
-
-	bio::filtering_streambuf<bio::input> out;
-#ifdef ZLIB_LIB_SUPPORTED
-	if (filter == CompressionFilter::GZIP) {
-		out.push(bio::gzip_decompressor());
-	}
-#endif
 #ifdef ZSTD_LIB_SUPPORTED
 	if (filter == CompressionFilter::ZSTD) {
-		out.push(bio::zstd_decompressor());
+		const char* src = reinterpret_cast<const char*>(data.begin());
+		size_t destSize = ZSTD_decompressBound(src, data.size());
```

Comment:
Again `makeString` can be used to allocate the StringRef of desired length and returned as is without an extra memcpy

Reply:
discussed in a separated thread